nat/service_ics.go

```diff
@@ -0,0 +1,147 @@
+// +build windows
+
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package nat
+
+import (
+	"net"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type serviceICS struct {
+	ifaces             map[string]struct{} // list in internal interfaces with enabled internet connection sharing
+	remoteAccessStatus string
+}
+
+func (nat *serviceICS) Enable() error {
+	status, err := powerShell("Get-Service RemoteAccess | foreach { $_.StartType }")
+	if err != nil {
+		return err
+	}
+	nat.remoteAccessStatus = string(status)
+
+	if _, err := powerShell("Set-Service -Name RemoteAccess -StartupType automatic"); err != nil {
+		return err
+	}
+	if _, err := powerShell("Start-Service -Name RemoteAccess"); err != nil {
+		return err
+	}
+
+	ifaceName, err := getPublicInterfaceName()
+	if err != nil {
+		return err
+	}
+
+	// Enable internet connection sharing for the public interface.
+	_, err = powerShell(`regsvr32 /s hnetcfg.dll;
+		$netShare = New-Object -ComObject HNetCfg.HNetShare;
+		$c = $netShare.EnumEveryConnection |? { $netShare.NetConnectionProps.Invoke($_).Name -eq "` + ifaceName + `" };
+		$config = $netShare.INetSharingConfigurationForINetConnection.Invoke($c);
+		$config.EnableSharing(0)`)
```

Comment:
`$config.EnableSharing(0)` 0 - enables sharing? :)

Reply:
Yep. 
`0` here is a type of sharing. 0 - public, 1 - private.

Reply:
Constants would improve readability here